diff --git a/lib/Redis/Client.php b/lib/Redis/Client.php
index 58bf96c..54ae758 100644
--- a/lib/Redis/Client.php
+++ b/lib/Redis/Client.php
@@ -119,9 +119,18 @@ class Redis_Client {
       global $conf;
 
       // Always prefer socket connection.
+      if (!empty($conf['redis_client_socket'])) {
+        $host = $conf['redis_client_socket'];
+        $port = NULL;
+      }
+      else {
+        $host = isset($conf['redis_client_host']) ? $conf['redis_client_host'] : self::REDIS_DEFAULT_HOST;
+        $port = isset($conf['redis_client_port']) ? $conf['redis_client_port'] : self::REDIS_DEFAULT_PORT;
+      }
+
       self::$_client = self::getClientInterface()->getClient(
-        isset($conf['redis_client_host']) ? $conf['redis_client_host'] : self::REDIS_DEFAULT_HOST,
-        isset($conf['redis_client_port']) ? $conf['redis_client_port'] : self::REDIS_DEFAULT_PORT,
+        $host,
+        $port,
         isset($conf['redis_client_base']) ? $conf['redis_client_base'] : self::REDIS_DEFAULT_BASE,
         isset($conf['redis_client_password']) ? $conf['redis_client_password'] : self::REDIS_DEFAULT_PASSWORD);
     }
